### PR TITLE
Rearrange matrix groups to reduce test build time

### PR DESCRIPTION
**Pull request by despo**

### CODE REVIEW

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -82,6 +82,11 @@
     end
   end
 
+  # Only run specs who configure the feature flag state once
+  if ENV['DEFAULT_FEATURE_FLAG_STATE'] != 'on'
+    config.filter_run_excluding :feature_flag
+  end
+
```
Comment (despo) [inline spec/rails_helper.rb:89]: Ideally we would be able to automate excluding the specs that set the state of the feature flag so we don't test twice (and add extra time to our test build), but this is the best alternative I can think of.

File: spec/middlewares/apply/content_security_policy_middleware_spec.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 require 'rails_helper'
 
-RSpec.describe Apply::ContentSecurityPolicyMiddleware, type: :request do
+RSpec.describe Apply::ContentSecurityPolicyMiddleware, type: :request, feature_flag: true do
```
Comment (despo) [inline spec/middlewares/apply/content_security_policy_middleware_spec.rb:3]: Tagging a spec

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -108,15 +108,25 @@ jobs:
...
+          - tests: unit_shared
             include-pattern: spec/**/**/*_spec.rb
-            exclude-pattern: spec/{system,requests}/**/*_spec.rb
-          - tests: integration
-            include-pattern: spec/{system,requests}/**/*_spec.rb
-            exclude-pattern: ''
+            exclude-pattern: spec/**/{candidate_interface,support_interface,provider_interface,referee_interface,*api*}/**/*_spec.rb,spec/system/**/*_spec.rb
+          - tests: unit_candidate-provider
+            include-pattern: spec/**/{candidate_interface,provider_interface}/**/*_spec.rb
+            exclude-pattern: spec/system/{candidate_interface,provider_interface}/**/*_spec.rb
```
Comment (cwrw) [inline .github/workflows/build.yml:119]: should we exclude `spec/system/**/*_spec.rb` here as well as we should always exclude system regardless?

File: spec/rails_helper.rb (hunk 1 above)
Comment (cwrw) [reply spec/rails_helper.rb:89]: I've had a long think about the feature flags bit. So if you set all the feature flags to on and run the tests, we are activating all the feature flags and in particular tests deactivating. The opposite of that would be to still run all tests and activate/deactive the specific features, but with the other features off as there is an edge-case where if we had one feature on (the one set in specific tests) and the other off we get a failure. I think that's why it might be best to not ignore those and not run them. Let me know your thoughts

**Review by cwrw — COMMENTED** [review]
The matrix looks good, thanks for spending time refining this! I left some thoughts on the feature flag bit though, how does the matrix times look if we don't ignore those?

**Review by cwrw — APPROVED** [review]
Looks great 🎉  This shaves off quite a bit, next step parallelise linting!
